### PR TITLE
update drawer layout and container logic

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -**
Updates Gradebook "Drawer" layout and "container" workflow to be more sensible.
The main goal of this PR is to clean up the Drawer component in order to get it tested.  However, the Drawer component as it existed was performing a weird set of end-runs in order to accomplish its goal, that could be much more easily accomplished with a bit of re-organization.

summary in ticket:
JIRA: [AU-8](https://openedx.atlassian.net/browse/AU-8)

[Diff of Drawer/index.jsx after becoming WithSidebar/index.jsx] (https://gist.github.com/muselesscreator/0da092234578fcf05386d3d24727fce3)
[Diff of Gradebook/index.jsx after becoming GradesTab/index.jsx] (https://gist.github.com/muselesscreator/9ef258522a0bf88cc0c29fd10b868a56)

**What changed?**

**Developer Checklist**
- [x] Test suites passing
- [x] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

[ How should a reviewer test this PR? ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [x] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: src/components/WithSidebar/index.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import selectors from 'data/selectors';
+import thunkActions from 'data/thunkActions';
+
+export class WithSidebar extends React.Component {
+  get sidebarClassNames() {
+    return classNames('sidebar', { open: this.props.open, 'd-none': this.props.isClosed });
+  }
+
+  get contentClassNames() {
+    return classNames('sidebar-contents', 'position-relative', {
```
Comment (leangseu-edx) [inline src/components/WithSidebar/index.jsx:29]: ❤️

**Review by leangseu-edx — APPROVED** [review]

File: src/data/actions/app.js [hunk 2]
```diff
@@ -12,11 +12,19 @@ const closeModal = createAction('closeModal');
  * @param {string} courseId - course ID from app context
  */
 const setCourseId = createAction('setCourseId');
+
+const filterMenu = StrictDict({
+  endTransition: createAction('filterMenu/endTransition'),
```
Comment (nsprenkle) [inline src/data/actions/app.js:17]: Is there a reason to action on start/end open? It seems needlessly actiony...

File: src/containers/GradebookPage/index.jsx [hunk 3]
```diff
@@ -1,45 +1,83 @@
...
+        parsed[key] = queryParams[key];
+      } else {
+        delete parsed[key];
+      }
+    });
+  }
+
+  render() {
+    return (
+      <WithSidebar
```
Comment (nsprenkle) [inline src/containers/GradebookPage/index.jsx:43]: Nit: I know we probably do this for layout reasons but I think it's weird to have a sidebar that's only logically connnected to the `GradesTab` at the `Page` level. Also the `WithSidebar` naming is a bit bizarre. I get what it's doing but my reaction is 🤨 every time I look at it.

File: src/components/WithSidebar/index.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
```
Comment (nsprenkle) [inline src/components/WithSidebar/index.jsx:1]: Esp. since this is a new component, docstrings explaining what these are and how to instantiate. Please and thanks.

File: src/containers/GradebookPage/index.jsx (hunk 3 above)
Comment (muselesscreator) [reply src/containers/GradebookPage/index.jsx:43]: With<Thing> is a pretty common React paradigm.  
In comparison to `<Drawer content=<Main content>>{sidebar content}</Drawer>`, it felt much cleaner.

as for the sidebar only being related to one tab... that alas is an existing issue.

File: src/data/actions/app.js (hunk 2 above)
Comment (muselesscreator) [reply src/data/actions/app.js:17]: because of how is called in the component.  All is for simpler components.

File: src/components/WithSidebar/index.jsx (hunk 4 above)
Comment (muselesscreator) [reply src/components/WithSidebar/index.jsx:1]: fair fair.  I am mostly planning a big wave of docstring adds as part of the next PR, but can add for this as it got re-worked.

**Comment by nsprenkle** [comment]
Replaced with https://github.com/edx/frontend-app-gradebook/pull/190
